src/nspawn/nspawn-mount.c

```diff
@@ -933,7 +926,13 @@ int mount_custom(
         for (i = 0; i < n; i++) {
                 CustomMount *m = mounts + i;
 
-                if (m->in_userns != in_userns)
+                if ((mount_settings & MOUNT_IN_USERNS) != m->in_userns)
+                        continue;
+
+                if (mount_settings & MOUNT_ROOT_ONLY && !path_equal(m->destination, "/"))
+                        continue;
+
+                if (mount_settings & MOUNT_NON_ROOT_ONLY && path_equal(m->destination, "/"))
```

Comment:
can you use the FLAG_SET() macro for these flags checks, I think this would make things a tiny bit more readable

Reply:
I took the liberty to apply `FLAGS-SET` everywhere where it was applicable in nspawn-mount.c